softwareComponents/configuration/include/configuration/rofibot.hpp

```diff
@@ -647,21 +679,27 @@ class Rofibot {
     }
 
     struct ModuleInfo {
-        ModuleInfo( const ModuleInfo& ) = default;
-        ModuleInfo( ModuleInfo&& o ) noexcept
-        : module( std::move( o.module ) ), inJointsIdx( std::move( o.inJointsIdx ) ),
-            outJointsIdx( std::move( o.outJointsIdx ) ), spaceJoints( std::move( o.spaceJoints ) ),
-            position( std::move( o.position ) )
-        {}
-
-        ModuleInfo( atoms::ValuePtr< Module >&& m, const std::vector< RoficomHandle >& i, const std::vector< RoficomHandle >& o,
-            const std::vector< SpaceJointHandle >& s, const std::optional< Matrix >& pos )
-        : module( std::move( m ) ), inJointsIdx( i ), outJointsIdx( o ), spaceJoints( s ), position( pos )
+        ModuleInfo( atoms::ValuePtr< Module > m, std::vector< RoficomHandle > i, std::vector< RoficomHandle > o,
+            std::vector< SpaceJointHandle > s, std::optional< Matrix > pos )
+        : module( std::move( m ) ), inJointsIdx( std::move( i ) ), outJointsIdx( std::move( o ) ),
+            spaceJoints( std::move( s ) ), position( std::move( pos ) )
         {}
 
-        ModuleInfo& operator=( const ModuleInfo& ) = default;
+        ModuleInfo( ModuleInfo&& o ) noexcept = default;
         ModuleInfo& operator=( ModuleInfo&& ) = default;
 
+        ModuleInfo( const ModuleInfo& o )
+        : ModuleInfo( o.module.clone(), o.inJointsIdx, o.outJointsIdx, o.spaceJoints, o.position )
+        {}
+        ModuleInfo& operator=( const ModuleInfo& o ) {
+            this->module = o.module.clone();
```

Comment:
@jajik You had a comment that you want to make the address of modules stable. You can manage that here.